File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/query/search/QueryDatabaseSearchCriteria.java
/*
 * Copyright 2014 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.query.search;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.AbstractObjectSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchCriteriaToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchOperator;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.query.id.IQueryDatabaseId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.query.search.QueryDatabaseSearchCriteria")
public class QueryDatabaseSearchCriteria extends AbstractObjectSearchCriteria<IQueryDatabaseId>
{

    private static final long serialVersionUID = 1L;

    public QueryDatabaseSearchCriteria()
    {
    }

    public QueryDatabaseSearchCriteria withOrOperator()
    {
        return (QueryDatabaseSearchCriteria) withOperator(SearchOperator.OR);
    }

    public QueryDatabaseSearchCriteria withAndOperator()
    {
        return (QueryDatabaseSearchCriteria) withOperator(SearchOperator.AND);
    }

    @Override
    protected SearchCriteriaToStringBuilder createBuilder()
    {
        SearchCriteriaToStringBuilder builder = super.createBuilder();
        builder.setName("QUERY_DATABASE");
        return builder;
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/query/search/QueryDatabaseSearchCriteria.js
/**
 * @author pkupczyk
 */
define([
  'require',
  'stjs',
  'as/dto/common/search/AbstractObjectSearchCriteria'
], function (require, stjs, AbstractObjectSearchCriteria) {
  var QueryDatabaseSearchCriteria = function () {
    AbstractObjectSearchCriteria.call(this)
  }
  stjs.extend(
    QueryDatabaseSearchCriteria,
    AbstractObjectSearchCriteria,
    [AbstractObjectSearchCriteria],
    function (constructor, prototype) {
      prototype['@type'] = 'as.dto.query.search.QueryDatabaseSearchCriteria'
      constructor.serialVersionUID = 1
    },
    {
      criteria: {
        name: 'Collection',
        arguments: ['ISearchCriteria']
      }
    }
  )
  return QueryDatabaseSearchCriteria
})
